stream_out.c

```diff
@@ -467,12 +467,13 @@ int out_create(struct audio_hw_device *dev,
     if ((xout->p_handle == NULL) || (!pcm_is_ready(xout->p_handle))) {
         ALOGE("%s() failed (%d). Can't open stream on device.", __FUNCTION__, errno);
         if (xout->p_handle != NULL) {
+            ALOGE("pcm error:'%s'", pcm_get_error(xout->p_handle));
             pcm_close(xout->p_handle);
             xout->p_handle = NULL;
         }
         *stream_out = NULL;
         free(xout);
-        return -ENOMEM;
+        return -errno;
```

Comment:
Consider the possibility of changing errno by free()

Reply:
Thanks. Fixed.